### PR TITLE
Adapt all components to `core.gardener.cloud/v1beta1` API group

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
Adapt all components to `core.gardener.cloud/v1beta1` API group.

**Release note**:

```action operator
All Gardener components now work with the `core.gardener.cloud/v1beta1` API group (instead of `core.gardener.cloud/v1alpha1`). Please make sure that all extension controllers can understand both the `v1alpha1` and `v1beta1` version to ensure a smooth update.
```

### CODE REVIEW

**Comment by rfranzke** [comment]
ping @timuthy @ialidzhikov @vpnachev

File: pkg/apis/core/v1beta1/constants/types_constants.go [hunk 1]
```diff
@@ -203,9 +219,9 @@ const (
 	LabelKubernetes = "kubernetes"
 	// LabelAPIServer is a constant for a label for the kube-apiserver.
 	LabelAPIServer = "apiserver"
-	// LabelControllerManager is a constant for a label for the kube-controller-manager.
+	// LabelAPIServer is a constant for a label for the kube-controller-manager.
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/constants/types_constants.go]: Changed by mistake?

File: pkg/apis/core/v1beta1/constants/types_constants.go [hunk 2]
```diff
@@ -203,9 +219,9 @@ const (
...
 	LabelControllerManager = "controller-manager"
-	// LabelScheduler is a constant for a label for the kube-scheduler.
+	// LabelAPIServer is a constant for a label for the kube-scheduler.
```
Comment (timuthy) [inline pkg/apis/core/v1beta1/constants/types_constants.go]: Changed by mistake?

**Review by timuthy — CHANGES_REQUESTED** [review]
Nice, just 2 nits

File: pkg/apis/core/v1beta1/constants/types_constants.go (hunk 1 above)
Comment (rfranzke) [reply pkg/apis/core/v1beta1/constants/types_constants.go]: Yes, nice catch, fixed, thanks.

File: pkg/apis/core/v1beta1/constants/types_constants.go (hunk 2 above)
Comment (rfranzke) [reply pkg/apis/core/v1beta1/constants/types_constants.go]: Yes, nice catch, fixed, thanks.

**Review by timuthy — APPROVED** [review]
/lgtm

**Comment by ialidzhikov** [comment]
/cc @vpnachev @swilen-iwanow 

> Please make sure that all extension controllers can understand both the `v1alpha1` and `v1beta1` version to ensure a smooth update.

**Pull request merged**